pkg/featureflags/featureflags_test.go

```
@@ -0,0 +1,376 @@
+	t.Cleanup(sh.Shutdown)
+}
+
+// ---- constructor ----------------------------------------------------------------
+
+func TestNewEmbeddedProvider_EmptyFilePath(t *testing.T) {
+	cfg := commoncfg.FeatureFlags{Enabled: true, FilePath: "", PollingInterval: 60 * time.Second}
+	_, err := NewEmbeddedProvider(cfg)
+	if err == nil {
+		t.Fatal("expected error for empty FilePath, got nil")
+	}
+	if err != ErrEmptyFilePath {
```

golangci-lint (2.12.2)

[error] comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error

(errorlint)